src/main/resources/public/ts/sniplets/paramServices.ts

```diff
@@ -206,8 +214,27 @@ export const paramServices = {
                 if (matiere && matiere.sous_matieres && matiere.sous_matieres.length > 0)
                     matiere.sous_matieres.forEach(sm => {
                         paramServices.that.subTopics.all.map(sb => {
```

Comment:
Tu fait un map mais tu ne récupère pas le résultat. Ce serait plutot un for each.